File: src/main/webapp/src/app/models/task.ts
export class Task {
  private _id: number;
  private _lessonId: number;
  private _title: string;
  private _description: string;
  private _maxPoints: number;
  private _isCodeTask: boolean;
  private _branchNamePrefix: string;

  constructor(id: number, lessonId: number, title: string, description: string, maxPoints: number, isCodeTask: boolean, branchNamePrefix: string) {
    this._id = id;
    this._lessonId = lessonId;
    this._title = title;
    this._description = description;
    this._maxPoints = maxPoints;
    this._isCodeTask = isCodeTask;
    this._branchNamePrefix = branchNamePrefix;
  }

  get id(): number {
    return this._id;
  }

  get lessonId(): number {
    return this._lessonId;
  }

  get title(): string {
    return this._title;
  }

  get description(): string {
    return this._description;
  }

  get maxPoints(): number {
    return this._maxPoints;
  }

  get isCodeTask(): boolean {
    return this._isCodeTask;
  }

  get branchNamePrefix(): string {
    return this._branchNamePrefix;
  }

  public static fromJSON(jsonObj: any): Task {
    return new Task(+jsonObj.id, +jsonObj.lessonId, jsonObj.title,
      jsonObj.description, +jsonObj.maxPoints, jsonObj.isCodeTask, jsonObj.branchNamePrefix);
  }

  public toJSON() {
    return {
      id: this.id,
      lessonId: this.lessonId,
      title: this.title,
      description: this.description,
      maxPoints: this.maxPoints,
      isCodeTask: this.isCodeTask,
      branchNamePrefix: this.branchNamePrefix
    }
  }
}

File: src/main/java/pl/coddlers/core/models/entity/Task.java
package pl.coddlers.core.models.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@ToString(exclude={"lesson", "submissions", "notes"})
public class Task {
    @Id
    @GeneratedValue
    private long id;

    @Column(nullable=false)
    private String title;

    private String description;

    @Column(nullable=false)
    private int maxPoints;

    private Boolean isCodeTask;

    private String branchNamePrefix;

    private Timestamp creationTime;

    @JsonIgnore
    @ManyToOne(targetEntity = Lesson.class)
    private Lesson lesson;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "task", targetEntity = Submission.class)
    private List<Submission> submissions = new ArrayList<>();

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "task", targetEntity = Note.class)
    private List<Note> notes = new ArrayList<>();
}
